sos/report/plugins/arcconf.py

```diff
@@ -21,11 +23,34 @@ class arcconf(Plugin, IndependentPlugin):
     commands = ("arcconf",)
 
     def setup(self):
-
-        # get list of adapters
-        self.add_cmd_output([
-            "arcconf getconfig 1",
-            "arcconf list",
-            "arcconf GETLOGS 1 UART"
-        ])
+        # Get the list of available adapters
+        listarcconf = self.collect_cmd_output("arcconf list")
+
+        # Parse the 'arcconf list' output and extract controller IDs.
+        # For each Controller ID found in 'arcconf list', add commands
+        # for getconfig and GETLOGS
+        #
+        # Sample 'arcconf list' output:
+        #
+        # Controller information
+        # -------------------------------------------------------------
+        #    Controller ID   : Status, Slot, Mode, Name, SerialNumber, WWN
+        # -------------------------------------------------------------
+        #    Controller 1:   : Optimal, Slot XXXX, XXXX, XXXX, XXXX, XXXX
+        # -------------------------------------------------------------
+        #    Controller 2:   : Optimal, Slot XXXX, XXXX, XXXX, XXXX, XXXX
+
+        if listarcconf['status'] == 0:
+            for line in listarcconf['output'].splitlines():
+                try:
+                    match = re.match(r"^[\s]*Controller (\d)+", line).group(0)
+                    controller_id = match.split()[1]
+                    if controller_id:
+                        # Add new commands with Controller ID
+                        self.add_cmd_output([
+                            f"arcconf getconfig {controller_id}",
+                            f"arcconf GETLOGS UART {controller_id}",
```

Comment:
The previous invocation of this was `arcconf GETLOGS $id UART` - does the position of the id not matter, or was it wrong previously?

I'm fine with whatever works, just something I noticed and want to double check since I'm not personally familiar with `arcconf`.

Reply:
Crucial observation. It does matter actually.

I have some special binary where it does not but in general it does.

Thanks @TurboTurtle , I will send updated patch.